### DISCUSSION TITLE
Loosing all historical data in energy dashboard after adding LUNA2000 batterie

### DISCUSSION THREAD

**Discussion by Mahai17** [wlcrs/huawei_solar#773] [General]
Hi,
I use this integration for more than one year. It is great.
I started without a batterie last year and used "sensor.inverter_daily_yield" for solar production in the energy dashboard. That worked fine. Now, I added a LUNA2000 batterie and I had to add the new entity (Input Yield = Daily Yield - Battery Day Discharge + Battery Day Charge). That works fine as well, but as a consequence, I loose all my historical solar production data in the energy dashboard. 
Is there any chance to preserve the historical data?

Martin Heiden

**Comment by Roving-Ronin** [comment]
I presume your saying g you added a custom sensor, using yaml?

Can you paste it below? What are you trying to g to achieve with this sensor btw ?

As you've described it, adding a custom sensor like this should have zero impact upon the existing inverter daily yield sensor and its data.

BTW also still a good idea to use the HA utility_meter component to take the lifetime yield sensor as the source and define a daily / weekly / monthly / yearly sensor for yield (as well as other things like grid import / export and house consumption).

**Comment by Mahai17** [comment]
Hi Roving,

before having the LUNA2000 I used sensor.inverter_daily_yield for PV Production in the HA Energy Dashboard.
After installing the LUNA2000 I have to use a new sensor, "sensor. solar_ertrag_heute" for PV Production in the HA Energy Dashboard. 
```
      solar_ertrag_heute:
        unit_of_measurement: 'kWh'
        device_class: energy
        value_template: >
          {{ (states('sensor.inverter_daily_yield') | float) - (states('sensor.battery_day_discharge') | float) + (states('sensor.battery_day_charge') | float ) }}
```
That worked fine. But as a consequence of the change I loose all the historical data in the HA Energy Dashboard.

And yes, thanks, I start using HA utility_meter component.

**Reply by Roving-Ronin** [reply]
Trying to follow what happened...

Think your saying that before you had Huawei, your OLD inverter reported using the sensor.inverter_daily_yield, and this sensor was then used in the Energy dashboard.

After updating to use Huawei inverter and using WLCRS integration, you found that it also uses the same sensor.inverter_daily_yield to report.

If this is correct, the Huawei / WLCRS created version of this sensor should be created as a duplicate, that would see if get the name 'sensor.inverter_daily_yield_2' (which is also the naming standard in the integration for if you have a second Huawei inverter... in which case you'd end up with a _3 instead, if you have 2 inverters).

________________

Options:

Look at exporting the Energy data from the History panel : https://www.home-assistant.io/integrations/history/

Then look at using something like these integrations to import it back, adding it to the Huawei inverter_daily_yield sensor:
https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data or https://github.com/klausj1/homeassistant-statistics

_____

Issues:

1) The formula that your using is not compatible with WLCRS v1.4.x naming format for the battery. If you have 1 batttery its' sensors will have battery_1_xxx as the sensor name. There will also be a 'batteries' sensor that is available, this is an alternative and shows the totals for ALL (be it 1 or 2) batteries attached to the inverter. I don't know of anyone with the $ for 3+ batteries, that would see at least 1 battery attached to a secondary inverter, but I assume that would produce a 'batteries_2' group of sensors.

For your formula to work it needs to be:

```
{{ (states('sensor.inverter_daily_yield') | float) - (states('sensor.battery_1_day_discharge') | float) + (states('sensor.battery_1_day_charge') | float ) }}
```

2) Bigger issue, that formula is (as best as I can put it)....of no use.  
If your after the 'Daily Yield', that is the amount of kWh Energy that the PV system produced from the sun for the day, you simply need to use the 'sensor.inverter_daily_yield'. If you have 2 inverters, in the Energy Dashboard IMHO you'd enter 'sensor.inverter_daily_yield_2' as a second source, allowing the Energy Dashboard to map out the production (and total) of the two inverters.

If you specifically want a seperate sensor that will track total yield for both inverters, then use:

```
  # Provides the combined daily yield of both inverters, in kWh.
  - sensor:
    - name: "Inverters - Daily Yield"
      unique_id: inverters_daily_yield
      unit_of_measurement: "kWh"
      device_class: "energy"
      state_class: total_increasing
      state: >-
        {% set inverter_1_daily_yield = states('sensor.inverter_daily_yield') | float(default=0) %}
        {% set inverter_2_daily_yield = states('sensor.inverter_daily_yield_2') | float(default=0) %}

        {% if states('sensor.inverter_daily_yield_2') == 'unknown' %}
          {{ inverter_1_daily_yield }}
        {% else %}
          {% set count = 0 %}
          {% set sum = 0 %}

          {% if states('sensor.inverter_daily_yield') != 'unknown' %}
            {% set sum = sum + inverter_1_daily_yield %}
            {% set count = count + 1 %}
          {% endif %}

          {% if states('sensor.inverter_daily_yield_2') != 'unknown' %}
            {% set sum = sum + inverter_2_daily_yield %}
            {% set count = count + 1 %}
          {% endif %}

          {% if count > 0 %}
            {{ sum }}
          {% else %}
            N/A
          {% endif %}
        {% endif %}
      availability: >
        {% set inverter_1_available = has_value('sensor.inverter_daily_yield') %}
        {% set inverter_2_available = false %}
        {% if 'sensor.inverter_daily_yield_2' in states %}
          {% set inverter_2_available = has_value('sensor.inverter_daily_yield_2') %}
        {% endif %}
        {{ inverter_1_available or inverter_2_available }}

```

If your trying to get a sensor that will show you the total consumption in the house, taking into account PV, Grid and Battery then use:

```
  # Provides the calculated current load, in Watts, of the house including power being drawn from solar, battery & grid.
  - sensor:
    - name: "House Consumption - Power"
      unique_id: house_consumption_power
      unit_of_measurement: "W"
      icon: mdi:home-lightning-bolt-outline
      device_class: "power"
      state_class: measurement
      state: >-
        {% set inverters_power = states('sensor.inverters_active_power') | float(default=0) %}
        {% set power_meter_power = states('sensor.power_meter_active_power') | float(default=0) %}

        {{ max(0, (inverters_power - power_meter_power)) }}

      availability: >-
        {{ states('sensor.inverters_active_power') | is_number and
           states('sensor.power_meter_active_power') | is_number }}

```

Also of use is looking at Nick Jensens 'PEES' that creates a bunch (like 300 !) sensors to track various power/energy flows. 
https://github.com/JensenNick/huawei_solar_pees

**Comment by Mahai17** [comment]
thank you very much. That's a lot of stuff. I will work it out and will come back.

**Comment by Mahai17** [comment]
Hi,

sorry for my bad English. 
I have an Huawei inverter and I use the WLCRS integration since April 2023. (worked fine).
Last week I installed the new LUNA2000 battery and it integrates automatically into WLCRS.
Two problems:

1. I installed the LUNA2000 before upgrading to the new version of WLCRS integration (I was not aware of the new version). I guess the old version had another naming convention, because the battery device was called battery_1. Then, I updated to 1.4.1. And because you are wondering about the entity names, I deinstalled and installed again the WLCRS integration. Now I have the correct device and entity names (il.e. sensor.batteries_state_of_capacity). However the integration still shows the inactive entities and the battery_1 device. Can I simply delete the old entities?

2. Home Assistant Energy Dashboard: Without the battery I used sensor.inverter_daily_yield as input for "Solar Production" and of course, that worked fine. Now I am confused. I followed your instruction in the WIKI and built a new sensor like "Input Yield = Daily Yield - Battery Day Discharge + Battery Day Charge".  And I used this new sensor for "Solar Production". That worked fine (without efficiency correction). Now, you say that is useless. However, I installed the pees package (great work). Now I use sensor.energy_yield_1 for "Solar Production". Yes, worked fine. Now coming back to my original question/problem. When I change the "Solar Production" input in the Energy Dashboard, I loose all historical PV Production data in the dashboard. But meanwhile I found a workaround. 

Long story short. I am happy. And many thanks for your work with this integration.

**Comment by Mahai17** [comment]
I can't remove the 11 "old" battery_1 entities. They are some kind of bind to the integration. Do you have a hint?
<img width="1212" alt="image" src="https://github.com/user-attachments/assets/b8be897b-9aa8-4200-aa3a-70aae2a47b66">

**Comment by Mahai17** [comment]
Just to make sure.
I deleted the integration and added again.
It adds 4 devices, see picture. is that correct? Battery_1 has only disabled entities.
<img width="593" alt="image" src="https://github.com/user-attachments/assets/34f28c16-772b-46c3-ad4e-77871d9444ab">

**Reply by Roving-Ronin** [reply]
Yes.
